sensors/src/main/scala/com/lynbrookrobotics/potassium/sensors/position/xyPositionCalculator.scala

```diff
@@ -0,0 +1,39 @@
+package com.lynbrookrobotics.potassium.sensors.position
+
+import com.lynbrookrobotics.potassium.units.{Point, Value3D}
+import com.lynbrookrobotics.potassium.{PeriodicSignal, Signal}
+import squants.Angle
+import squants.space._
+import squants.time.{Seconds, Time}
+
+/**
+  * Calculates the current xy position given the current distance traveled since
+  * start of calculations and the current angle
+  * @param initPosition initial position
+  * @param angle trigonometric angle, where 0 is defined at the x axis
+  * @param totalDistanceTraveled signal for total distance traveled
+  */
+class xyPositionCalculator(initPosition: Point,
+                           angle: PeriodicSignal[Angle],
+                           totalDistanceTraveled: PeriodicSignal[Length]) {
+  def this(angle: PeriodicSignal[Angle],
+           totalDistanceTraveled: PeriodicSignal[Length]) = {
+    this(new Point(Feet(0), Feet(0)), angle, totalDistanceTraveled)
+  }
+
+  private val initAngle = angle.currentValue(Seconds(0))
+  private val previousAngle = angle.scanLeft(initAngle)((previous, _, _) => previous)
+  private var accumPose: Point = initPosition
+
+  private val deltaDistanceTraveled = totalDistanceTraveled.scanLeft(Feet(0)){
```

Comment:
So I am trying to get a signal for the difference between the angle traveled between computations. So If we moved from 8ft to 10ft since the last computation, I want this to return 2ft